src/dmd/dtoh.d

```diff
@@ -0,0 +1,1703 @@
+            }
+        }
+        else
+        {
+            if (p.defaultArg)
+            {
+                //printf("%s %d\n", p.defaultArg.toChars, p.defaultArg.op);
+                //return;
+                buf.writestring("/*");
+                buf.writestring(" = ");
+                buf.writestring(p.defaultArg.toChars());
+                //p.defaultArg.accept(this);
+                buf.writestring("*/");
+            }
+        }
+    }
+
+    override void visit(AST.Expression e)
+    {
+        debug (Debug_DtoH)
+        {
+            printf("[AST.Expression enter] %s\n", e.toChars());
+            scope(exit) printf("[AST.Expression exit] %s\n", e.toChars());
+        }
+        assert(0);
+    }
+
+    override void visit(AST.NullExp e)
+    {
+        debug (Debug_DtoH)
+        {
+            printf("[AST.NullExp enter] %s\n", e.toChars());
+            scope(exit) printf("[AST.NullExp exit] %s\n", e.toChars());
+        }
+        buf.writestring("_d_null");
+    }
+
+    override void visit(AST.ArrayLiteralExp e)
+    {
+        debug (Debug_DtoH)
+        {
+            printf("[AST.ArrayLiteralExp enter] %s\n", e.toChars());
+            scope(exit) printf("[AST.ArrayLiteralExp exit] %s\n", e.toChars());
+        }
+        buf.writestring("arrayliteral");
+    }
+
+    override void visit(AST.StringExp e)
+    {
+        debug (Debug_DtoH)
+        {
+            printf("[AST.StringExp enter] %s\n", e.toChars());
+            scope(exit) printf("[AST.StringExp exit] %s\n", e.toChars());
+        }
+        assert(e.sz == 1 || e.sz == 2);
+        if (e.sz == 2)
+            buf.writeByte('L');
+        buf.writeByte('"');
+
+        for (size_t i = 0; i < e.len; i++)
+        {
+            uint c = e.charAt(i);
+            switch (c)
+            {
+                case '"':
+                case '\\':
+                    buf.writeByte('\\');
+                    goto default;
+                default:
+                    if (c <= 0xFF)
+                    {
+                        if (c <= 0x7F && isprint(c))
+                            buf.writeByte(c);
+                        else
+                            buf.printf("\\x%02x", c);
+                    }
+                    else if (c <= 0xFFFF)
+                    {
+                        buf.printf("\\x%02x\\x%02x", c & 0xFF, c >> 8);
+                    }
+                    else
+                    {
+                        buf.printf("\\x%02x\\x%02x\\x%02x\\x%02x",
+                                   c & 0xFF, (c >> 8) & 0xFF, (c >> 16) & 0xFF, c >> 24);
+                    }
+                    break;
+            }
+        }
+        buf.writeByte('"');
+    }
+
+    override void visit(AST.RealExp e)
+    {
+        debug (Debug_DtoH)
+        {
+            printf("[AST.RealExp enter] %s\n", e.toChars());
+            scope(exit) printf("[AST.RealExp exit] %s\n", e.toChars());
+        }
+        buf.writestring("0");
```

Comment:
This looks a bit suspicious, its OK if you want to defer this to a later PR though.

Reply:
Thats definitely wrong:

```
extern(C++) struct S
{
    double member = 3.141;
}
```
```
struct S
{
    _d_double member;
    S() : member(0) {}
};
```

But I would prefer to keep the initial version close to the original PRs functionality.

Reply:
Ok. Please put a large TODO next then.

Reply:
Done